main.ts

```diff
@@ -174,21 +262,92 @@ class Translator {
     this.visit(expr);
   }
 
-  /** Searches for super() calls and emits an initializer for it if found. */
-  maybeEmitInitializerListForSuperCall(body: ts.Block) {
-    if (!body || body.statements.length === 0) return;
+  /**
+   * Handles constructor initializer lists and bodies.
+   *
+   * <p>Dart's super() ctor calls have to be moved to the constructors initializer list, and `const`
```

Comment:
Nope, this is just handling the initializer stuff, ctor inheritance is a different can of worms.